lgc/builder/InOutBuilder.cpp

```diff
@@ -287,22 +287,23 @@ void InOutBuilder::markGenericInputOutputUsage(bool isOutput, unsigned location,
   auto resUsage = getPipelineState()->getShaderResourceUsage(m_shaderStage);
 
   // Mark the input or output locations as in use.
-  std::map<unsigned, unsigned> *inOutLocMap = nullptr;
+  std::map<InOutLocationInfo, InOutLocationInfo> *inOutLocInfoMap = nullptr;
```

Comment:
Consider to use SmallDenseMap

I think we should use LLVM native containers instead of STL.

Reply:
Other maps in InOutUsage are use std::map, should we align with them?
Btw, a little bit concern that the LLVM programmer's manual said "DenseMap allocates space for a large number of key/value pairs (it starts with 64 by default), it will waste a lot of space if your keys or values are large". I'm not sure the number [0~65535] of `InOutLocationInfo` as key/value can be seen as large or small?

Reply:
> Other maps in InOutUsage are use std::map, should we align with them?

Ok. Let's keep it as-is.

> Btw, a little bit concern that the LLVM programmer's manual said "DenseMap allocates space for a large number of key/value pairs (it starts with 64 by default), it will waste a lot of space if your keys or values are large"

 I assume SmallDenseMap with 32 allocated slots (MaxInOutLocCount) is OK for us to use. Let's keep it a TODO item for future work.

Reply:
Ok. Thanks.